sync_known_issues.py

```diff
@@ -144,51 +141,47 @@ def __init__(self, config, squad_project, strict=False):
                 # ignore projects that are not defined
                 # in the SquadProject.projects
                 self.projects.remove(project)
-                if strict:
-                    raise SquadKnownIssueException("Project not defined: %s" % project)
+                raise SquadKnownIssueException("Project not defined: %s" % project)
         self.environments = set()
         for item in config.get('environments'):
-            if 'slug' in item.keys():
-                if item['slug'] in squad_project.environment_slugs:
-                    self.environments.add(item['slug'])
-                else:
-                    if strict:
-                        raise SquadKnownIssueException(
-                            "Incorrect environment: %s" % item['slug'])
-            if 'architecture' in item.keys():
-                if item['architecture'] in squad_project.environment_architectures.keys():
-                    for env in squad_project.environment_architectures[item['architecture']]:
-                        self.environments.add(env)
-                else:
-                    if strict:
-                        raise SquadKnownIssueException(
-                            "Unknown architecture: %s" % item['architecture'])
+            if item in squad_project.environments:
+                self.environments.add(item)
+            else:
+                raise SquadKnownIssueException(
+                    "Incorrect environment: %s" % item['slug'])
+
+    def __repr__(self):
```

Comment:
wouldn't it be better to use yaml for object serialization?

Reply:
Sure, that's a good idea.

Reply:
Fixed in https://github.com/danrue/qa-reports-known-issues/commit/aceefbae6697d228e03dc213ed2d5c158ff294c1 (i guess github is having some delay in updating the pull request, according to their status page)